### PR TITLE
add statestore health check endpoint.

### PR DESCRIPTION
**Pull request by moonchanyong**
# Description

I  have added a health check endpoint for statestore.
but I have not changed component-contrib yet.
below code is an expected change.

```go
// github.com/dapr/components-contrib/state/store.go

type Store interface {
	BulkStore
	Init(metadata Metadata) error
	Features() []Feature
	Delete(req *DeleteRequest) error
	Get(req *GetRequest) (*GetResponse, error)
	Set(req *SetRequest) error
	Ping() error
}
```

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #2167

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: pkg/http/api.go [hunk 1]
```diff
@@ -317,6 +317,12 @@ func (a *api) constructHealthzEndpoints() []Endpoint {
 			Version: apiVersionV1,
 			Handler: a.onGetHealthz,
 		},
+		{
+			Methods: []string{fasthttp.MethodGet},
+			Route:   "healthz/{storeName}",
```
Comment (yaron2) [inline pkg/http/api.go:322]: This needs to be `healthz/state/{storeName}` to allow for other components to have health checks.

In addition, we probably need a gRPC equivalent for this.

**Review by yaron2 — CHANGES_REQUESTED** [review]

File: pkg/http/api.go (hunk 1 above)
Comment (moonchanyong) [reply pkg/http/api.go]: @yaron2  I have added changes about gRPC and URI. PTAL.

**Comment by moonchanyong** [comment]
@yaron2
I have changed this pr.  this pr will fail the test because component-contrib pr not is merged yet.

File: pkg/actors/actors_test.go [hunk 2]
```diff
@@ -101,6 +101,10 @@ func (f *fakeStateStore) Init(metadata state.Metadata) error {
 	return nil
 }
 
+func (f *fakeStateStore) Ping() error {
+	return nil
```
Comment (daixiang0) [inline pkg/actors/actors_test.go:105]: Is it possible to add a error case?

**Comment by daixiang0** [comment]
ping @moonchanyong

**Comment by pkedy** [comment]
ping @moonchanyong

I'm a little late on the components PR but I think having a component health check is not limited to state stores. It might be better to create a separate interface that the runtime can assert all components with.

```golang
// Somewhere in components-contrib

package health

type Pinger interface {
     Ping() error
}
```

```golang
// From the API

if pinger, ok := component.(health.Pinger); ok {
    err := pinger.Ping()
    // Handler error case
} else {
    // Handle case where a health check is not implemented
}
```

**Comment by moonchanyong** [comment]
@pkedy  
ok, I submit a new pr with `pinger` interface to component-contrib.

**Comment by moonchanyong** [comment]
@yaron2 

Hi, I pushed new changes. PTAL.

**Pull request closed without merging**